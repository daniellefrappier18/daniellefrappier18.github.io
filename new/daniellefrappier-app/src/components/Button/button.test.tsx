import React from "react";
import { Button } from "./button.component";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("Button component", () => {
  // test component with props
  it("Button renders correctly Props", () => {
    const tree = renderer
      .create(
        <Button
          buttonType="primary"
          size="medium"
          href="https://www.daniellefrappier.com"
          target="_blank"
          isFullWidth={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button renders correctly with small size and secondary", () => {
    const tree = renderer
      .create(
        <Button
          buttonType="secondary"
          size="small"
          href="https://www.daniellefrappier.com"
          target="_blank"
          isFullWidth={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button renders correctly with no props", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button renders correctly at full width", () => {
    const tree = renderer.create(<Button isFullWidth={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  afterEach(cleanup);
});
