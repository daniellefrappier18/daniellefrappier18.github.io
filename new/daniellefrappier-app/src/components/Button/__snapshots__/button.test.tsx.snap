// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button renders correctly Props 1`] = `
<a
  className="buttonstyle__ButtonComponent-sc-ypiqrg-0 fUGPhs"
  href="https://www.daniellefrappier.com"
  size="medium"
  target="_blank"
/>
`;

exports[`Button component Button renders correctly at full width 1`] = `
<button
  className="buttonstyle__ButtonComponent-sc-ypiqrg-0 jVyhAE"
  role="button"
  size="medium"
/>
`;

exports[`Button component Button renders correctly with no props 1`] = `
<button
  className="buttonstyle__ButtonComponent-sc-ypiqrg-0 idFOlP"
  role="button"
  size="medium"
/>
`;

exports[`Button component Button renders correctly with small size and secondary 1`] = `
<a
  className="buttonstyle__ButtonComponent-sc-ypiqrg-0 nczYq"
  href="https://www.daniellefrappier.com"
  size="small"
  target="_blank"
/>
`;

exports[`Button component Button renders with no props 1`] = `
<button
  className="buttonstyle__ButtonComponent-sc-ypiqrg-0 idFOlP"
  role="button"
  size="medium"
/>
`;
